### PR TITLE
Registry benchmark

### PR DESCRIPTION
**Pull request by vjgaur**

### CODE REVIEW

File: pallets/registry/src/benchmarking.rs [hunk 1]
```diff
@@ -62,15 +62,189 @@ benchmarks! {
...
+		let digest = <T as frame_system::Config>::Hashing::hash(&registry[..]);
+		let id_digest = <T as frame_system::Config>::Hashing::hash(
+			&[&registry.encode()[..], &did.encode()[..]].concat()[..],
+		);
+		let registry_id = Ss58Identifier::to_registry_id(&(id_digest).encode()[..]).unwrap();
+
+		Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+
+		let registry_archive: Vec<u8> = (2u8..u8::MAX).cycle().take(T::MaxEncodedRegistryLength::get().try_into().unwrap()).collect();
+		let utx_registry: InputRegistryOf::<T> = BoundedVec::try_from(registry_archive)
```
Comment (smohan-dw) [inline pallets/registry/src/benchmarking.rs]: registry_ archive and utx_registry variables are not required here

File: pallets/registry/src/benchmarking.rs [hunk 2]
```diff
@@ -62,15 +62,189 @@ benchmarks! {
...
+    let registry: InputRegistryOf::<T> = BoundedVec::try_from(raw_registry)
+        .expect("Test Registry should fit into the expected input length of the test runtime");
+    let id_digest = <T as frame_system::Config>::Hashing::hash(
+        &[&registry.encode()[..], &did.encode()[..]].concat()[..],
+    );
+    let registry_id = Ss58Identifier::to_registry_id(&(id_digest).encode()[..]).unwrap();
+
+    Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+
+    let origin_creator =  <T as Config>::EnsureOrigin::generate_origin(caller.clone(), did.clone());
```
Comment (smohan-dw) [inline pallets/registry/src/benchmarking.rs]: ?? not required

File: pallets/registry/src/benchmarking.rs [hunk 3]
```diff
@@ -62,15 +62,189 @@ benchmarks! {
...
+        )).encode()[..]
+    ).expect("Authorization ID should be generated");
+
+    let authorization = <Authorizations<T>>::get(&authorization_id).expect("Authorization should exist");
+
+    assert_last_event::<T>(Event::AddAuthorization {
+        registry: registry_id,
+        authorization: authorization_id,
+        delegate: delegate,
+    }.into());
```
Comment (smohan-dw) [inline pallets/registry/src/benchmarking.rs:164]: assert_last_event would be a good enough check, as it checks for the call success. We can remove the storage checks

File: pallets/registry/src/benchmarking.rs [hunk 4]
```diff
@@ -62,15 +62,189 @@ benchmarks! {
...
+}
+
+add_delegate {
+    let l in 1 .. MAX_REGISTRY_SIZE;
```
Comment (smohan-dw) [inline pallets/registry/src/benchmarking.rs]: Registry size does not matter here. We need to use the max_delegates one

File: pallets/registry/src/benchmarking.rs [hunk 5]
```diff
@@ -62,15 +62,189 @@ benchmarks! {
...
+verify {
+    assert_last_event::<T>(Event::AddAuthorization {
+        registry: registry_id,
+        authorization: authorization_id,
+        delegate,
+    }.into());
+}
+
+remove_delegate {
+    let l in 1 .. MAX_REGISTRY_SIZE;
```
Comment (smohan-dw) [inline pallets/registry/src/benchmarking.rs]: same as above

File: pallets/registry/src/benchmarking.rs [hunk 6]
```diff
@@ -62,15 +62,189 @@ benchmarks! {
...
+    Pallet::<T>::archive(origin.clone(), registry_id.clone()).expect("Should archive the registry");
+
+}: _<T::RuntimeOrigin>(origin, registry_id.clone())
+
+verify {
+    assert_last_event::<T>(Event::Restore { registry: registry_id, authority: did }.into());
+}
+
+add_admin_delegate {
+    let l in 1 .. MAX_REGISTRY_SIZE;
```
Comment (smohan-dw) [inline pallets/registry/src/benchmarking.rs]: Registry size does not matter here. We need to use the max_delegates one

**Review by smohan-dw — CHANGES_REQUESTED** [review]

File: pallets/registry/src/benchmarking.rs (hunk 1 above)
Comment (vjgaur) [reply pallets/registry/src/benchmarking.rs]: Done

File: pallets/registry/src/benchmarking.rs (hunk 2 above)
Comment (vjgaur) [reply pallets/registry/src/benchmarking.rs]: Done

File: pallets/registry/src/benchmarking.rs (hunk 3 above)
Comment (vjgaur) [reply pallets/registry/src/benchmarking.rs:164]: okay

Comment (vjgaur) [reply pallets/registry/src/benchmarking.rs:173]: Done

**Review by smohan-dw — DISMISSED** [review]

**Review by smohan-dw — APPROVED** [review]

**Pull request merged**